src/gg.regression.unity.bots/Editor/Scripts/ActionManager/RGActionAnalysis.cs

```diff
@@ -872,47 +863,30 @@ public override void VisitMemberAccessExpression(MemberAccessExpressionSyntax no
                                     return;
                             }
-                            string[] path = GetActionPathFromSyntaxNode(objectType, node);
-                            AddAction(new MouseButtonAction(path, objectType, RGActionParamFunc<int>.Constant(mouseButton)), node);
+                            string[] path = GetActionPathFromSyntaxNode(node);
+                            AddAction(new MouseButtonAction(path, null, RGActionParamFunc<int>.Constant(mouseButton)), node);
                         } else if (typeof(DeltaControl).IsAssignableFrom(exprType))
                         {
                             if (propSym.Name == "scroll")
                             {
                                 // Mouse.current.scroll
-                                var classDecl = node.Ancestors().OfType<ClassDeclarationSyntax>().FirstOrDefault();
-                                if (classDecl == null) return;
-                                Type objectType = FindType(_currentModel.GetDeclaredSymbol(classDecl));
-                                if (!typeof(MonoBehaviour).IsAssignableFrom(objectType))
-                                {
-                                    AddAnalysisWarning(node, "Inputs handled outside of a MonoBehaviour are not supported");
-                                    return;
-                                }
-                                string[] path = GetActionPathFromSyntaxNode(objectType, node);
-                                AddAction(new MouseScrollAction(path, objectType), node);
+                                string[] path = GetActionPathFromSyntaxNode(node);
+                                AddAction(new MouseScrollAction(path, null), node);
                             }
                         }
                     }
                 } else if (type == typeof(UnityEngine.InputSystem.Pointer))
                 {
-                    var classDecl = node.Ancestors().OfType<ClassDeclarationSyntax>().FirstOrDefault();
-                    if (classDecl == null) return;
-                    Type objectType = FindType(_currentModel.GetDeclaredSymbol(classDecl));
-                    if (!typeof(MonoBehaviour).IsAssignableFrom(objectType))
-                    {
-                        AddAnalysisWarning(node, "Inputs handled outside of a MonoBehaviour are not supported");
-                        return;
-                    }
-
-                    string[] path = GetActionPathFromSyntaxNode(objectType, node);
+                    string[] path = GetActionPathFromSyntaxNode(node);
                     switch (propSym.Name)
                     {
                         case "position":
                         case "delta":
                         {
-                            AddAction(new MousePositionAction(path, objectType), node);
+                            AddAction(new MousePositionAction(path, null), node);
```

Comment:
I notice the second parameter is almost always `null`, are there cases where it isn't?

Reply:
The `AddAction` method resolves the object type later - I made this change because I was previously duplicating the same code over and over for resolving the object type, and also because we need to handle the possibility of the target object not being known yet if this code is from a helper method.

See the following code for where the resolution of ObjectType happens:
https://github.com/Regression-Games/RGUnityBots/blob/5c2dda94d8b2e73542f78ad09fa0849cea500b7e/src/gg.regression.unity.bots/Editor/Scripts/ActionManager/RGActionAnalysis.cs#L991-L1000